app/views/api/integrations/apicast/shared/_proxy_endpoints.html.slim

```diff
@@ -4,14 +4,14 @@
 = f.inputs "API gateway", id: 'proxy-endpoints', data: {apicast_custom_urls: apicast_custom_urls?}, class: "apicast-only-settings #{'hidden' if is_service_mesh}", disabled: is_service_mesh do
   = f.input :sandbox_endpoint, label: "Staging Public Base URL",
             input_html: { class: 'proxy-endpoint',
-                          data: { default: (proxy.default_staging_endpoint_apiap ) },
+                          data: { default: (proxy.default_staging_endpoint unless proxy.self_managed?) },
```

Comment:
I was thinking... This will cause the default to be null when the service is self-managed, what currently doesn't happen with all providers on APIAP.

Another option could be renaming `Proxy#deployment_strategy_apiap` to `Proxy#hosted_service_deployment_strategy` and here use `proxy.hosted_service_deployment_strategy.default_staging_endpoint` (and proxy.hosted_service_deployment_strategy.default_production_endpoint below)